app/src/main/java/ch/epfl/skysync/screens/CalendarUI.kt

```diff
@@ -3,136 +3,194 @@ package ch.epfl.skysync.screens
-        AvailabilityStatus.MAYBE -> Color.Blue
-        AvailabilityStatus.NO -> Color.Red
-        AvailabilityStatus.UNDEFINED -> Color.Gray
-      }
-
+fun showTile(
+    date: LocalDate,
+    slot: TimeSlot,
+    scaleHeight: Float,
+    scaleWidth: Float,
+    viewModel: UserViewModel,
+    index: Int
+) {
+  var colorUp by remember {
+    mutableStateOf(
+        availabilityToColor(viewModel.user.value.availabilities.getAvailabilityStatus(date, slot)))
+  }
+  colorUp =
+      availabilityToColor(viewModel.user.value.availabilities.getAvailabilityStatus(date, slot))
   Box(
       modifier =
-          Modifier.size(size).background(backgroundColor).clickable {
-            availabilityStatus = nextAvailabilityStatus(date, slot)
-          })
+          Modifier.fillMaxHeight(scaleHeight)
+              .testTag(date.toString() + slot.toString())
+              .fillMaxWidth(scaleWidth)
+              .background(color = colorUp, shape = RoundedCornerShape(0.dp))
+              .clickable {
+                colorUp =
+                    availabilityToColor(
+                        viewModel.user.value.availabilities.nextAvailabilityStatus(date, slot))
+              })
 }
-
+/**
+ * Composable function to display the calendar view.
+ *
+ * @param today The current date to initialize the calendar.
+ * @param viewModel user viewmodel (used to determine availabilities status)
+ */
 @Composable
-fun showCalendarAvailabilities() {
-  Calendar { date, slot, size -> showTile(date, slot, size) }
+fun showCalendarAvailabilities(today: LocalDate, viewModel: UserViewModel) {
+  Calendar(today, viewModel)
+}
+/** Preview function to display the calendar view. */
+@Composable
+@Preview
+fun CalendarPreview() {
+  var viewModel = UserViewModel.createViewModel(firebaseUser = null)
+  viewModel.user.value.availabilities.setAvailabilityByDate(
+      LocalDate.now(), TimeSlot.AM, AvailabilityStatus.MAYBE)
+  showCalendarAvailabilities(LocalDate.now(), viewModel)
 }
-
 /**
  * Composable function to display a calendar view.
  *
- * @param modularShow a Composable function that what to display each tile in the calendar depending
- *   on the day and if it's AM/PM (AM=0:00 and PM=12:00).
+ * @param today The current date to initialize the calendar.
+ * @param viewModel user viewmodel (used to determine availabilities status)
  */
 @Composable
-fun Calendar(modularShow: @Composable (LocalDate, TimeSlot, Dp) -> Unit) {
-  var currentWeekStartDate by remember { mutableStateOf(getStartOfWeek(LocalDate.now())) }
+fun Calendar(today: LocalDate, viewModel: UserViewModel) {
+  var currentWeekStartDate by remember { mutableStateOf(getStartOfWeek(today)) }
 
   Column(
       modifier = Modifier.fillMaxSize().padding(16.dp),
   ) {
-    WeekView(currentWeekStartDate, modularShow)
+    WeekView(currentWeekStartDate, viewModel)
     Spacer(modifier = Modifier.height(8.dp))
     Row {
+      Spacer(modifier = Modifier.width(138.dp))
```

Comment:
You use pixel instead of percentage, because if the screen size change the UI may overflow